### DISCUSSION TITLE
make launch error

### DISCUSSION THREAD

**Discussion by nikolatopic131** [carla-simulator/carla#7237] [Q&A] [answered]
Hello. I'm not sure if this has been asked before, I scrolled a few pages but couldn't find anything. I am trying to install carla from source on windows 10 and I was following these instructions: https://carla.readthedocs.io/en/latest/build_windows/ . I followed the instructions exactly as they are step by step, but on the final step when running the "make launch" command I am getting the error as shown in the screenshot. I tried to delete the Build file inside my carla directory and run make PythonAPI again but it didn't help, I restarted my PC and tried running make launch again and then got error from the second screenshot. Anyone knows how to solve this?
![error_make_launch](https://github.com/carla-simulator/carla/assets/162899513/90ae7e5f-3b54-4996-9fc1-19c0f67570b9)
![error_make_launch2](https://github.com/carla-simulator/carla/assets/162899513/a144cf33-fffb-4e54-a3d4-0332f153225b)

**Comment by bg-sumu** [comment]
Could you copy paste the Error in Text ?

**Reply by nikolatopic131** [reply]
Hello. Sorry for the late reply, I wasn't at home. 
Here is the error text: Building 7 actions with 20 processes...
  [1/7] UE4Editor-StreetMapRuntime.dll
  LINK : fatal error LNK1181: cannot open input file 'C:\carla\Unreal\CarlaUE4\Plugins\Carla\CarlaDependencies\lib\osm2odr.lib'
  [2/7] UE4Editor-CarlaUE4.dll
  LINK : fatal error LNK1181: cannot open input file 'C:\carla\Unreal\CarlaUE4\Plugins\Carla\CarlaDependencies\lib\osm2odr.lib'
  [3/7] UE4Editor-Carla.dll
  LINK : fatal error LNK1181: cannot open input file 'C:\carla\Unreal\CarlaUE4\Plugins\Carla\CarlaDependencies\lib\osm2odr.lib'
  [4/7] Module.CarlaTools.cpp
  C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\um\winnt.h(603): warning C4005: 'TEXT': macro redefinition
  C:\UnrealEngine\Engine\Source\Runtime\Core\Public\HAL/Platform.h(1082): note: see previous definition of 'TEXT'
  C:\carla\Unreal\CarlaUE4\Plugins\Carla\Source\Carla\Vehicle\WheeledVehicleMovementComponentNW.h(120): warning C4996: 'UWheeledVehicleMovementComponent': PhysX is deprecated. Use the UChaosWheeledVehicleMovementComponent fron the ChaosVehiclePhysics Plugin. Please update your code to the new API before upgrading to the next release, otherwise your project will no longer compile.
  C:\carla\Unreal\CarlaUE4\Plugins\Carla\Source\Carla\Vehicle\WheeledVehicleMovementComponentNW.h(148): warning C4996: 'UVehicleWheel': PhysX is deprecated. Use the UChaosVehicleWheel from the ChaosVehiclePhysics Plugin. Please update your code to the new API before upgrading to the next release, otherwise your project will no longer compile.
  C:\carla\Unreal\CarlaUE4\Plugins\Carla\Source\Carla/Vehicle/CarlaWheeledVehicle.h(72): warning C4996: 'AWheeledVehicle': PhysX is deprecated. Use the AWheeledVehiclePawn from the ChaosVehiclePhysics Plugin. Please update your code to the new API before upgrading to the next release, otherwise your project will no longer compile.
  C:\carla\Unreal\CarlaUE4\Plugins\Carla\Source\Carla/Vehicle/CarlaWheeledVehicle.h(188): warning C4996: 'UWheeledVehicleMovementComponent4W': PhysX is deprecated. Use the UChaosWheeledVehicleMovementComponent from the ChaosVehiclePhysics Plugin. Please update your code to the new API before upgrading to the next release, otherwise your project will no longer compile.

  You have included windows.h before MinWindows.h
  All useless stuff from the windows headers won't be excluded !!!

  C:\UnrealEngine\Engine\Source\Runtime\Core\Public\Windows/MinWindows.h(44): warning C4005: 'NOMINMAX': macro redefinition
  C:\carla\Unreal\CarlaUE4\Plugins\Carla\CarlaDependencies\include\boost/asio/detail/config.hpp(1566): note: see previous definition of 'NOMINMAX'
  C:/carla/Unreal/CarlaUE4/Plugins/CarlaTools/Source/CarlaTools/Private/Online/CustomFileDownloader.cpp(11): fatal error C1083: Cannot open include file: 'OSM2ODR.h': No such file or directory
make: *** [CarlaUE4Editor] Error 6

**Comment by bg-sumu** [comment]
Go through this
https://github.com/carla-simulator/carla/discussions/7237#discussioncomment-8778366

**Reply by nikolatopic131** [reply]
I'm not sure if the link you posted is working correctly. It doesn't lead me anywhere, just sends me back to the top of this post. Would you mind sharing the link again, assuming it was a link to a post with a solution?

**Comment by bg-sumu** [comment]
Hey i found a similar issue and i had sent it here as it had a solution as well. Idk what happened to the link
Anyways one more similar issue https://github.com/carla-simulator/carla/discussions/6918

**Comment by nikolatopic131** [comment, marked as answer]
I found the solution on a youtube video. Here it is:

The issue is in the file BuildOSM2ODR.bat that's under carla\Util\BuildTools.
In line 112
cmake -G %GENERATOR% %PLATFORM%^
should be changed to
cmake -G %GENERATOR% -A x64^

**Reply by xavisolesoft** [reply]
Hi nikolatopic131,

Thank you for sharing.

Could you tell me what generator are you using to dive deep on this solution? It is Ninja? Visual Studio 2019?
